src/NULS/TransactionBuilder.h

```diff
@@ -0,0 +1,120 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "../proto/NULS.pb.h"
+
+#include <algorithm>
+
+namespace TW::NULS {
+
+struct TransactionBuilder {
+    /// Plans a transaction by selecting UTXOs and calculating fees.
+    static Proto::TransactionPlan plan(const NULS::Proto::TransactionPurpose& purpose) {
```

Comment:
Could you please add a test case for sending max amount?

Reply:
Nuls don't have sending max amount limit.
I add transaction max data size check.

Reply:
oh, consider this case: user wants to send all UTXOs, we have logic to adjust output count and fees